llvm/lib/Target/DirectX/DXILShaderFlags.cpp

```diff
@@ -84,15 +129,21 @@ void ComputedShaderFlags::print(raw_ostream &OS) const {
 /// Return the shader flags mask of the specified function Func.
 const ComputedShaderFlags &
 ModuleShaderFlags::getFunctionFlags(const Function *Func) const {
-  const auto Iter = llvm::lower_bound(
-      FunctionFlags, Func,
-      [](const std::pair<const Function *, ComputedShaderFlags> FSM,
-         const Function *FindFunc) { return (FSM.first < FindFunc); });
+  auto Iter = FunctionFlags.find(Func);
   assert((Iter != FunctionFlags.end() && Iter->first == Func) &&
-         "No Shader Flags Mask exists for function");
+         "Get Shader Flags : No Shader Flags Mask exists for function");
   return Iter->second;
 }
 
+/// Merge specified shader flags mask SF with current mask of the specified
+/// function Func.
+void ModuleShaderFlags::mergeFunctionShaderFlags(const Function *Func,
+                                                 ComputedShaderFlags SF) {
+  assert(FunctionFlags.contains(Func) &&
+         "Merge Shader Flags : No Shader Flags Mask exists for function");
+  FunctionFlags[Func].merge(SF);
+}
```

Comment:
This doesn't appear to be used.

Reply:
> This doesn't appear to be used.

Deleted unused function `mergeFunctionShaderFlags()`.